test/intfuncs.jl

```diff
@@ -11,6 +11,12 @@
 
 @test typeof(gcd(Int32(3), Int32(15))) == Int32
 
+@test gcd(typemax(Int), 1) == 1
+@test gcd(-typemax(Int), 1) == 1
+@test gcd(typemin(Int), 1) == 1
+divex = sizeof(Int) == 4 ? OverflowError : InexactError
+@test_throws divex gcd(typemin(Int), typemin(Int))
```

Comment:
I don't think it makes sense to be too specific about the exceptions that are returned. If we want to specify the exact exception types, then there would be quite a few routines that would need to catch exceptions thrown internally and then re-throw another exception, which is presumably too expensive. As a result, the various exceptions thrown by arithmetic and conversion routines are neither well documented nor well designed.

What I'm really looking for is a way to say "this operation throws an exception" in the test system; I don't care which it is.